resources/qml/Preferences/ProfilesPage.qml

```diff
@@ -43,14 +43,22 @@ Item
     }
 
     property var currentItemName: hasCurrentItem ? base.currentItem.name : ""
+    property var currentItemDisplayName: hasCurrentItem ? base.qualityManagementModel.getQualityItemDisplayName(base.currentItem) : ""
 
     property var isCurrentItemActivated:
     {
         if (!base.currentItem)
         {
             return false;
         }
-        return base.currentItem.name == Cura.MachineManager.activeQualityOrQualityChangesName;
+        if (base.currentItem.is_read_only)
```

Comment:
Why do you need a different behavior if the item is read_only?

Reply:
Because the functions (or APIs) for selecting a custom profile and a built-in quality/intent profile are different.